app/src/main/java/ca/llamabagel/transpo/data/SearchRepository.kt

```diff
@@ -25,47 +31,46 @@ object OttawaBoundaries {
     const val MAX_LNG = -75.351405
 }
 
-class SearchRepository @Inject constructor(private val database: TransitDatabase, private val strings: StringsGen) {
+@Singleton
+@ExperimentalCoroutinesApi
+class SearchRepository @Inject constructor(
+    private val database: TransitDatabase,
+    private val strings: StringsGen,
+    private val dispatcher: CoroutinesDispatcherProvider
+) {
 
-    suspend fun getSearchResults(query: String): List<SearchResult> = withContext(Dispatchers.IO) {
-        if (query.isEmpty()) return@withContext emptyList<SearchResult>()
+    val routeChannel = ConflatedBroadcastChannel<List<SearchResult.RouteItem>>()
```

Comment:
```suggestion
    private val routeChannel = ConflatedBroadcastChannel<List<SearchResult.RouteItem>>()
    val routeFlow get() = routeChannel.asFlow()
```

I think it's best to keep the channels themselves private and expose a public flow property instead. This will also need to be updated in the `TripsRepository` at some point. (I'll handle that change).